### ISSUE TITLE
CPC label

### ISSUE THREAD

**Issue by jacky123465** [andi611/Self-Supervised-Speech-Pretraining-and-Representation-Learning#47]
Hello, Author ! Thank you for your sharing ! 

I have some problems about the CPC label. I find that code in 'dataloader.py'.

`assert('train-clean-100' in sets and len(sets) == 1) `

Does this code mean that I can only use the 'train-clean-100' ? 

In addition, I have use kaldi to align the LibriSpeech but I can not get the file 'converted_aligned_phone.txt'. How can I obtain the CPC label of other datasets ?

Would you like to answer my questions? Thank you very much !

**Comment by andi611** [comment]
> Hello, Author ! Thank you for your sharing !
> 
> I have some problems about the CPC label. I find that code in 'dataloader.py'.
> 
> `assert('train-clean-100' in sets and len(sets) == 1) `
> 
> Does this code mean that I can only use the 'train-clean-100' ?

Yes, because in the [CPC paper](https://arxiv.org/pdf/1807.03748.pdf) Section 3.1 it says:
"For audio, we use a 100-hour subset of the publicly available LibriSpeech dataset [30]. Although the
dataset does not provide labels other than the raw text, we obtained force-aligned phone sequences with the Kaldi toolkit [31] and pre-trained models on Librispeech. We have made the aligned phone labels and our train/test split available for download on Google Drive."

Hence, the `train-clean-100` is split into train/test sets. Here we follow the setting of DeepMind.

> In addition, I have use kaldi to align the LibriSpeech but I can not get the file 'converted_aligned_phone.txt'. How can I obtain the CPC label of other datasets ?
> 

The file `converted_aligned_phone.zip` is provided in this repo under `data/cpc_phone`. See [this section](https://github.com/andi611/Self-Supervised-Speech-Pretraining-and-Representation-Learning#kaldi-phone-set-recommended) of README.

I hope this helps!

**Comment by jacky123465** [comment]
Thank for your answer! 

However, I also have another question: I would like to use 'train-clean-360'  or ''train-other-500 to start my experiments. How can I obtain the  'converted_aligned_phone.txt' of these datasets.

**Comment by andi611** [comment]
> Thank for your answer!
> 
> However, I also have another question: I would like to use 'train-clean-360' or ''train-other-500 to start my experiments. How can I obtain the 'converted_aligned_phone.txt' of these datasets.

The file `converted_aligned_phone.txt` is generated with Kaldi, and provided by DeepMind in the CPC paper. I simply downloaded the file from DeepMind's google drive, I do not have experience with this.
